File: Linked Stack/python/stack.py
from node import Node
class Stack:
    def __init__(self) -> None:
        self.stackTop:Node = None
        
    def push(self, item):
        temp = Node()
        temp.item = item
        temp.next = self.stackTop
        self.stackTop = temp
    
    def pop(self):
        if(self.isEmpty()): raise Exception("Stack is Empty")
        temp = self.stackTop
        item = temp.item
        self.stackTop = temp.next
        del temp
        return item
    
    def display(self):
        temp = self.stackTop
        print("[", end="")
        while (temp != None):
            print(temp.item, end="")
            temp = temp.next
            print(","if temp != None else "", end="")
        print("]")
    
    def getTop(self):
        if(self.isEmpty()): raise Exception("Stack is Empty")
        return self.stackTop.item
    
    def isEmpty(self):
        return self.stackTop == None

File: Linked Stack/dart/stack.dart
import 'dart:io';

import 'node.dart';

class Stack<T> {
  Node? stackTop;
  Stack() {
    stackTop = null;
  }

  void push(dynamic itme) {
    Node node = Node();
    node.itme = itme;
    node.next = stackTop;
    stackTop = node;
  }

  dynamic pop() {
    if (isEmpty()) throw Exception("Stack is Empty");
    Node? temp = stackTop!;
    dynamic itme = temp.itme;
    stackTop = null;
    stackTop = temp.next;
    temp = null;
    return itme;
  }

  dynamic getTop() {
    if (isEmpty()) throw Exception("Stack is Empty");
    return stackTop?.itme;
  }

  void display() {
    Node? temp = stackTop;

    stdout.write("[");
    while (temp != null) {
      stdout.write("${temp.itme}");
      temp = temp.next;
      stdout.write("${temp == null ? "" : ","}");
    }
    stdout.writeln("]");
  }

  bool isEmpty() {
    return stackTop == null;
  }
}
